lib/server/services/lhcPeriod/LhcPeriodStatisticsService.js

```diff
@@ -27,23 +27,33 @@ const { lhcPeriodStatisticsAdapter } = require('../../../database/adapters');
  */
 class LhcPeriodStatisticsService {
     /**
-     * Find or create a lhc period model by its name or id
+     * Find a lhc period model by its name or id
      * @param {LhcPeriodIdentifier} identifier the criteria to find run type
      * @return {Promise<SequelizeLhcPeriod[]>} the run type found or null
      */
-    async getById({ id, name }) {
-        return lhcPeriodStatisticsAdapter.toEntity(await LhcPeriodStatisticsRepository.findOne({ id, name }));
+    async getByIdentifier({ id, name }) {
+        const queryBuilder = dataSource.createQueryBuilder();
+        if (id) {
+            queryBuilder.where('id').is(id);
+        }
+        if (name) {
+            queryBuilder.where('name').is(name);
+        }
+        queryBuilder.include({ association: 'lhcPeriodStatistics' });
+
+        const lhcPeriod = await LhcPeriodRepository.findOne(queryBuilder);
+        return lhcPeriod ? lhcPeriodAdapter.toEntity(lhcPeriod) : null;
```

Comment:
Adapting from sequelize is the work of high level services (for now), we try to keep it all in the same place